### PR TITLE
Refresh plugin for 2023

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -2,6 +2,6 @@
   <extension>
     <groupId>io.jenkins.tools.incrementals</groupId>
     <artifactId>git-changelist-maven-extension</artifactId>
-    <version>1.3</version>
+    <version>1.4</version>
```
Comment (basil) [inline .mvn/extensions.xml:5]: Latest version.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,6 +1,4 @@
 // NOTE: Cannot use ACI agents because build requires `unzip` to be available
 buildPlugin(useContainerAgent: true, configurations: [
-        [ platform: "linux", jdk: "8" ],
-        [ platform: "windows", jdk: "8" ],
         [ platform: "linux", jdk: "11" ]
```
Comment (basil) [inline Jenkinsfile:3]: Java 11 is needed as of 2.361. Tests do not yet pass on Java 17 for this plugin and I did not investigate why.

File: declarative-pipeline-migration-assistant/pom.xml [hunk 3]
```diff
@@ -26,6 +26,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins.to-declarative</groupId>
       <artifactId>declarative-pipeline-migration-assistant-api</artifactId>
+      <version>${project.version}</version>
```
Comment (basil) [inline declarative-pipeline-migration-assistant/pom.xml:29]: No reason to have this dependency managed in the parent POM when this is the only consumer. Simpler to inline the version.

File: declarative-pipeline-migration-assistant/pom.xml [hunk 4]
```diff
@@ -48,28 +49,12 @@
...
-    <dependency>
-      <groupId>org.apache.commons</groupId>
-      <artifactId>commons-lang3</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>joda-time</groupId>
-      <artifactId>joda-time</artifactId>
-      <version>2.10.6</version>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>commons-lang3-api</artifactId>
```
Comment (basil) [inline declarative-pipeline-migration-assistant/pom.xml:53]: Preferring the library plugin to an inline JAR for linkage efficiency.

File: declarative-pipeline-migration-assistant/pom.xml [hunk 5]
```diff
@@ -48,28 +49,12 @@
...
-      <version>3.9.0</version>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.commons</groupId>
-      <artifactId>commons-lang3</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>joda-time</groupId>
-      <artifactId>joda-time</artifactId>
-      <version>2.10.6</version>
```
Comment (basil) [inline declarative-pipeline-migration-assistant/pom.xml:67]: Unused and therefore deleted.

File: declarative-pipeline-migration-assistant/pom.xml [hunk 6]
```diff
@@ -48,28 +49,12 @@
...
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>commons-lang3-api</artifactId>
     </dependency>
     <dependency>
       <groupId>org.jenkinsci.plugins</groupId>
       <artifactId>pipeline-model-definition</artifactId>
-      <version>1.9.2</version>
```
Comment (basil) [inline declarative-pipeline-migration-assistant/pom.xml:72]: Managed by the BOM so deleted.

File: declarative-pipeline-migration-assistant/pom.xml [hunk 7]
```diff
@@ -93,31 +78,28 @@
...
       <groupId>com.coravy.hudson.plugins.github</groupId>
       <artifactId>github</artifactId>
-      <version>1.34.1</version>
+      <version>1.36.1</version>
       <optional>true</optional>
     </dependency>
     <dependency>
       <groupId>org.6wind.jenkins</groupId>
       <artifactId>lockable-resources</artifactId>
-      <version>2.14</version>
```
Comment (basil) [inline declarative-pipeline-migration-assistant/pom.xml:102]: Managed by the BOM so deleted.

File: declarative-pipeline-migration-assistant/pom.xml [hunk 8]
```diff
@@ -93,31 +78,28 @@
...
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>build-timeout</artifactId>
-      <version>1.18</version>
+      <version>1.25</version>
       <optional>true</optional>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>config-file-provider</artifactId>
-      <version>3.7.1</version>
```
Comment (basil) [inline declarative-pipeline-migration-assistant/pom.xml:114]: Managed by the BOM so deleted.

File: declarative-pipeline-migration-assistant/pom.xml [hunk 9]
```diff
@@ -93,31 +78,28 @@
...
       <optional>true</optional>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>htmlpublisher</artifactId>
-      <version>1.29</version>
```
Comment (basil) [inline declarative-pipeline-migration-assistant/pom.xml:120]: Managed by the BOM so deleted.

File: pom.xml [hunk 10]
```diff
@@ -88,41 +86,11 @@
...
-      <dependency>
-        <groupId>org.codehaus.plexus</groupId>
-        <artifactId>plexus-utils</artifactId>
-        <version>3.3.0</version>
-      </dependency>
-      <dependency>
-        <groupId>io.atlassian.fugue</groupId>
-        <artifactId>fugue</artifactId>
-        <version>4.7.2</version>
-      </dependency>
```
Comment (basil) [inline pom.xml:125]: Unneeded and probably just there to satisfy Enforcer from a time before the BOM existed.

**Comment by basil** [comment]
Used my admin powers to replay the build with the `Jenkinsfile` changes: https://ci.jenkins.io/job/Plugins/job/declarative-pipeline-migration-assistant-plugin/job/PR-166/4/

**Comment by olamy** [comment]
doh with those changes build time jump locally from 7 minutes to 10 minutes! 
same on ci.j.io https://ci.jenkins.io/job/Plugins/job/declarative-pipeline-migration-assistant-plugin/job/master/buildTimeTrend
is it a known issue?

**Comment by basil** [comment]
Sorry, not sure offhand.

**Pull request merged**